Capstone/Autocorrect/AutocorrectFunctions.m

```diff
@@ -4,17 +4,36 @@
 //
 //  Created by Abel Kelbessa on 7/14/22.
 //
-
 #import "AutocorrectFunctions.h"
 #include "math.h"
+#import "College.h"
 
 @implementation AutocorrectFunctions
 + (instancetype)shared {
     AutocorrectFunctions *shared= [[AutocorrectFunctions alloc] init];
     return shared;
 }
 
+- (NSString *)findCorrectWord:(NSString *)searchWord :(College *)colleges {
+    double minDiffereceBetweenWords = 1000;
+    NSString *correctWord;
+    for (College *college in colleges) {
+        double hammingDistance = [self hammingDistanceCalculator:searchWord : [[college.name componentsSeparatedByString:@" "] objectAtIndex:0]];
+        double hammingDistanceFullCollegeName = [self hammingDistanceCalculator:searchWord : college.name];
+        if (hammingDistance < minDiffereceBetweenWords) {
+            if (hammingDistanceFullCollegeName < hammingDistance) {
+                minDiffereceBetweenWords = hammingDistanceFullCollegeName;
+            } else {
+                minDiffereceBetweenWords = hammingDistance;
+            }
+            correctWord = college.name;
+        }
+    }
+    return correctWord;
+}
+
 - (void)createCoordinates {
```

Comment:
nit but try these functions aren't quite self-explanatory so its important to add brief comments that explain the logic.